app/src/components/panels/SupportProposal.js

```diff
@@ -0,0 +1,197 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import BN from 'bn.js'
+import { useAragonApi } from '@aragon/api-react'
+import {
+  Button,
+  ButtonBase,
+  Field,
+  GU,
+  Info,
+  TextInput,
+  useSidePanelFocusOnReady,
+  useTheme,
+} from '@aragon/ui'
+import { toDecimals, round, pct } from '../../lib/math-utils'
+import useAccountTotalStaked from '../../hooks/useAccountTotalStaked'
+import { formatTokenAmount } from '../../lib/token-utils'
+
+const SupportProposal = React.memo(function SupportProposal({ id, onDone }) {
+  const theme = useTheme()
+  const [amount, setAmount] = useState({
+    value: '0',
+    valueBN: new BN(0),
+  })
+
+  const {
+    api,
+    appState: { stakeToken },
+  } = useAragonApi()
+  const inputRef = useSidePanelFocusOnReady()
+
+  const totalStaked = useAccountTotalStaked()
+  const nonStakedTokens = stakeToken.balanceBN.sub(totalStaked)
+
+  const handleEditMode = useCallback(
+    editMode => {
+      setAmount(amount => ({
+        ...amount,
+        value: amount.valueBN.gte(0)
```

Comment:
Should we move this bit of code to an useMemo so is cleaner to read which is this value? have the feeling that is a bit noisy all this inside the  object

Reply:
Maybe we can save the value in a var and not inline it directly, but note that we cannot use a `useMemo` since this is all done inside a `useCallback`